Sources/TuistGenerator/Generator/DerivedFileGenerator.swift

```diff
@@ -14,12 +14,11 @@ protocol DerivedFileGenerating {
     /// - Throws: An error if the generation of the derived files errors.
     /// - Returns: A project that might have got mutated after the generation of derived files, and a
     ///     function to be called after the project generation to delete the derived files that are not necessary anymore.
-    func generate(graph: Graph, project: Project, sourceRootPath: AbsolutePath) throws -> (Project, [SideEffectDescriptor])
+    func generate(graph: Graph, project: Project, sourceRootPath: AbsolutePath) throws -> (Project, Set<SideEffectDescriptor>)
```

Comment:
for a separate PR - we can try modeling this as a `GeneratedInfoPlistGraphMapper` - it was left in the generator to ease the migration to descriptors and we didn't have the mapper concept ironed out quite yet then :) 

That should help simplify this (and avoid it attempting to mutate the project mid generation!)